bench/micro/Bench/Database/LSMTree/Internal/Run/BloomFilter.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE NumericUnderscores #-}
+{- HLINT ignore "Use camelCase" -}
+{- HLINT ignore "Eta reduce" -}
+
+module Bench.Database.LSMTree.Internal.Run.BloomFilter (benchmarks) where
+
+import           Criterion.Main
+import qualified Data.BloomFilter.Easy as Bloom.Easy
+import           Data.Foldable (Foldable (..))
+import           Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import           Data.Word
+import           Database.LSMTree.Extras
+import           Database.LSMTree.Internal.Run.BloomFilter as Bloom
+import           System.Random
+import           System.Random.Extras
+import           Test.QuickCheck (generate, shuffle)
+
+-- See 'utxoNumPages'.
+benchmarks :: Benchmark
+benchmarks = bgroup "Bench.Database.LSMTree.Internal.Run.BloomFilter" [
+      bgroup "elems" [
+          env (elemEnv 0.1 2_500_000 1_000_000 0) $ \ ~(b, xs) ->
+            bench "onlyTruePositives 0.1" $ whnf (elems b) xs
+        , env (elemEnv 0.9 2_500_000 1_000_000 0) $ \ ~(b, xs) ->
+            bench "onlyTruePositives 0.9" $ whnf (elems b) xs
+        , env (elemEnv 0.1 2_500_000 0 1_000_000) $ \ ~(b, xs) ->
+            bench "onlyNegatives 0.1" $ whnf (elems b) xs
+        , env (elemEnv 0.9 2_500_000 0 1_000_000) $ \ ~(b, xs) ->
+            bench "onlyNegatives 0.9" $ whnf (elems b) xs
+        ]
+    , bgroup "construction" [
+          env (constructionEnv 2_500_000) $ \ m ->
+            bench "incrementalST 0.1" $ whnf (constructBloom mkBloomST 0.1) m
+        , env (constructionEnv 2_500_000) $ \ m ->
+            bench "incrementalST 0.9" $ whnf (constructBloom mkBloomST 0.9) m
+        , env (constructionEnv 2_500_000) $ \ m ->
+            bench "easyList 0.1" $ whnf (constructBloom mkBloomEasy 0.1) m
+        , env (constructionEnv 2_500_000) $ \ m ->
+            bench "easyList 0.9" $ whnf (constructBloom mkBloomEasy 0.9) m
```

Comment:
We could reuse the same `env` for all 4 runs

Reply:
Yes.